waves/src/WalletInfo.tsx

```diff
@@ -29,8 +30,11 @@ interface WalletInfoProps {
 export default function WalletInfo({ balance }: WalletInfoProps) {
     const { isOpen, onOpen, onClose } = useDisclosure();
     const btnRef = React.useRef(null);
-    const address = "LTB_addressxtqwseasdas"; // TODO get address from
     const [withDrawAddress, setWithDrawAddress] = React.useState("");
+    const [address, setAddress] = React.useState("LTB_addressxtqwseasdas");
+    getAddress().then((address) => {
+        setAddress(address);
+    });
```

Comment:
This should be inside a `useEffect` otherwise it will run on every re-render.

Reply:
If you want to make this less stateful, you could also use something like this: https://docs.react-async.com/guide/async-components#more-flexibility-with-useasync

In that case, you would probably want to extract a component for the whole QR code that internally renders something (or nothing) while the `getAddress` is pending.

Reply:
The same pattern can likely be applied in other cases as well!

Reply:
Cool stuff, used it in the last patch: [`86d3141` (#31)](https://github.com/comit-network/droplet/pull/31/commits/86d31416072d2ad684dea6035ddfa23d1e80c1b9)